src/components/ResourceList/components/FilterControl/components/DateSelector/DateSelector.tsx

```diff
@@ -273,15 +273,15 @@ class DateSelector extends React.PureComponent<CombinedProps, State> {
     if (newOption === DateFilterOption.OnOrBefore) {
       onFilterKeyChange(filterMaxKey);
       onFilterValueChange(
-        selectedDate ? formatDateValue(selectedDate) : undefined,
+        selectedDate ? formatDateForTimezone(selectedDate) : undefined,
       );
       return;
     }
 
     if (newOption === DateFilterOption.OnOrAfter) {
       onFilterKeyChange(filterMinKey);
       onFilterValueChange(
-        selectedDate ? formatDateValue(selectedDate) : undefined,
+        selectedDate ? formatDateForTimezone(selectedDate) : undefined,
```

Comment:
Maybe you could store the result of `formatDateForTimezone` into a constant since you are using it twice?